mergify_engine/actions/merge_base.py

```diff
@@ -698,9 +698,17 @@ async def merge_report(
             # Strict mode has been enabled in branch protection but not in
             # mergify
             conclusion = check_api.Conclusion.FAILURE
-            title = "Branch protection setting 'strict' conflicts with Mergify configuration"
+            title = "Branch protection setting 'strict' conflicts with Mergify configuration."
             summary = ""
 
+        elif (
+            await self._is_branch_protection_linear_history_enabled(ctxt)
+            and self.config["method"] == "merge"
+        ):
+            conclusion = check_api.Conclusion.FAILURE
+            title = "Branch protection setting 'linear history' conflicts with Mergify configuration."
+            summary = "Branch protection setting 'linear history' works only if `method=squash` or `method=rebase`."
```

Comment:
```suggestion
            summary = "Branch protection setting 'linear history' works only if `method: squash` or `method: rebase`."
```
At least use the same syntax than the YAML.

Do we need a test for this?